### PR TITLE
Adapt to changes needed for `gardener-node-agent`

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area os
/kind enhancement
/os garden-linux

**What this PR does / why we need it**:
This PR adapts this extension to changes needed for `gardener-node-agent`, see https://github.com/gardener/gardener/pull/8647 for more information.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/8023

**Special notes for your reviewer**:
- [x] Depends on https://github.com/gardener/gardener/pull/8647 and a new release containing this PR, hence this PR remains in draft state until then.

Kindly use the dedicated commits for an easier review process.

/cc @oliver-goetz @MrBatschner @danielfoehrKn 

**Release note**:

```noteworthy operator
This extension is now prepared to run with an enabled `UseGardenerNodeAgent` feature gate.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
PR is ready for review now that g/g@v1.82 is released

**Comment by rfranzke** [comment]
I plan to add a few unit tests for the new `actuator.go` file early next week.

**Comment by rfranzke** [comment]
OK, @oliver-goetz @MrBatschner @danielfoehrKn PTAL :)

**Comment by rfranzke** [comment]
/hold
Let's wait with the merge for another revendoring with `gardener/gardener@v1.83.0` containing https://github.com/gardener/gardener/pull/8707

File: pkg/controller/operatingsystemconfig/actuator.go
Comment (MrBatschner) [inline pkg/controller/operatingsystemconfig/actuator.go:204]:
```suggestion
	// add scripts and dropins for containerd
```

File: pkg/controller/operatingsystemconfig/generator/generator.go
Comment (MrBatschner) [inline pkg/controller/operatingsystemconfig/generator/generator.go:71]: Maybe I am not fully understanding the flow and/or intention but in https://github.com/gardener/gardener-extension-os-gardenlinux/blob/e9ab9eeda3623b13715959f4b103bd1af0c9041f/pkg/controller/operatingsystemconfig/actuator.go#L88, the data for CloudInit is generated by the `handleProvisionOSC()` function, yet here, if `useGardenerNodeAgent` is not used, the old template based generator is used. From what I can see, the actual output seems to be the same for both generating paths.
Can't we do away with the templater and use `handleProvisionOSC()` and `handleReconcileOSC()` exclusively for both cases?

**Review by MrBatschner — COMMENTED** [review]
see inline

File: pkg/controller/operatingsystemconfig/generator/generator.go
Comment (rfranzke) [reply pkg/controller/operatingsystemconfig/generator/generator.go:71]: Maybe we could, but I would prefer to keep things separate for now so that the old legacy generator can be easily deleted later when `gardener-node-agent` is GA and the `cloud-config-downloader` scenario fully disappeared. This way, we don't change anything for the current world (no `gardener-node-agent` in place), but just for the new feature when the feature gate is turned on. Hence, if anything goes wrong, it's just for the new scenario, not for the old one.

We still need to run the generator anyways for the migration scenario (when the `UseGardenerNodeAgent` feature gate is turned on for the first time) where `cloud-config-downloader` still runs on existing nodes. It needs to download the new user data (containing the `gardener-node-agent` systemd unit), so we wouldn't gain much from trying to harmonize anything now here.

File: pkg/controller/operatingsystemconfig/actuator.go
Comment (danielfoehrKn) [inline pkg/controller/operatingsystemconfig/actuator.go:65]: Could you add a comment here that after the migration to gardener-node-agent is done & gardener-node-agent feature gate is removed in the gardenlet, we only need to return the extra units and files? (then we can do `return nil, nil, nil, nil, extensionUnits, extensionFiles, err`)

**Review by danielfoehrKn — COMMENTED** [review]
Need to rework #133  after this PR is merged. 
Looks good to me. I like that the gardener-node-agent simplifies the extensions even more (generation of the execution script with systemd units & files is done not in each extension, but on the node via the gardener-node-agent)

**Review by danielfoehrKn — APPROVED** [review]

**Review by MrBatschner — APPROVED** [review]
/lgtm

**Comment by rfranzke** [comment]
Can we release this pR?

**Pull request merged**
